File: app/src/main/java/com/example/urlsenderapp/MainActivity.java
package com.example.urlsenderapp;

import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;
import android.os.Looper;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        Toast.makeText(this, "Sender app running in background", Toast.LENGTH_SHORT).show();

        Intent serviceIntent = new Intent(this, BackgroundSenderService.class);
        startService(serviceIntent);

        new Handler(Looper.getMainLooper()).postDelayed(() -> {
            finish();
        }, 2000);
    }
}

File: app/src/main/java/com/example/urlsenderapp/MainActivity.kt
package com.example.urlsenderapp

import android.os.Bundle
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {

    private lateinit var urlSender: UrlSender
    private lateinit var logTextView: TextView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        logTextView = findViewById(R.id.logTextView)

        urlSender = UrlSender(this) { log ->
            runOnUiThread {
                logTextView.append("\n$log")
            }
        }

        urlSender.startSending()
    }
}

